### DISCUSSION TITLE
How to get the full path name such as a 'c:/a/b/c.vtk'.

### DISCUSSION THREAD

**Discussion by timeanddoctor** [Kitware/trame#191] [Q&A]
How to get the full path name such as a 'c:/a/b/c.vtk'.

I used the ClientFile but not succeed.
from trame.app.file_upload import ClientFile

**Comment by jourdain** [comment]
[When a file is chosen using an `<input type="file">`, the real path to the source file is not shown in the input's value attribute for obvious security reasons.](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#notes)

**Comment by timeanddoctor** [comment]
Thanks. 
If opening the .obj model of my client, I need upload it to the server firstly in trame?

**Reply by jourdain** [reply]
Yes if you aim to process and/or render it on the server side. 

That is basically what you are currently doing and that's why with that `ClientFile` you have access to its content + name directly but not the path. In other word, the upload already happened.
